jaxrs/src/main/java/org/killbill/billing/jaxrs/json/CatalogJson.java

```diff
@@ -112,6 +113,17 @@ public CatalogJson(final StaticCatalog catalog) throws CatalogApiException {
             final PlanJson planJson = new PlanJson(plan);
             productJson.getPlans().add(planJson);
         }
+        // Add products that are not part of any plan
+        for (final Product product : XMLproducts) {
+            final String productName = product.getName();
+            ProductJson productJson = productMap.get(product.getName());
+            if (!productMap.containsKey(productName)) {
+                if (productJson == null) {
```

Comment:
is this `if` condition required? If the `productMap` does not contain the `productName`, wouldn't the `productJson` be null?